### DISCUSSION TITLE
Duplicate onClick handlers added (and type error)

### DISCUSSION THREAD

**Discussion by mattsbennett** [hugocxl/react-echarts#44] [General]
This handler is called multiple times - it seems with each render, another instance of the handler is added. Also, there is no type definition for `onClick` it seems.

```
onClick={(params: any) => handleChartClick(params)}
```

**Comment by mattsbennett** [comment]
Added related issue: https://github.com/hugocxl/react-echarts/issues/45

**Comment by hugocxl** [comment]
@mattsbennett about types in event handlers: echarts itselt is not a typescript-first lib, and I could not find specific types declarations for event handlers within its source code. In fact, I had to declare a list with the possible events handlers to be able to pass them as props. I hate having to do this but I could not find a better option. I came back over this issue to see if latest versions would fix this issue but it's still unresolved. 

Any ideas on how to overcome this issue is welcomed.

**Reply by mattsbennett** [reply]
I found some types for `onDataZoom`, namely

```
import { DataZoomPayloadBatchItem } from "echarts/types/src/component/dataZoom/helper.js"
import { Payload } from "echarts"
```

(the params returned to the handler may be either of these types). However, I'm not convinced the typescript gymnastics required are worthwhile at this point.
